addon/components/imgix-image.js

```diff
@@ -155,17 +155,22 @@ export default Component.extend(ResizeAware, {
 
       merge(theseOptions, get(this, 'options'));
 
-      for (let param of get(this, '_pathAsUrl.searchParams')) {
-        set(theseOptions, param[0], param[1]);
-      }
-
       if (get(config, 'APP.imgix.debug')) {
         merge(theseOptions, get(this, '_debugParams'));
       }
 
+      Object.keys(theseOptions).forEach((key) => {
+        if (!pathAsUri.hasQueryParam(key)) {
+          pathAsUri.addQueryParam(key, theseOptions[key]);
+        }
+      });
+
       return get(this, '_client').buildURL(
-        get(this, '_pathAsUrl.pathname'),
-        theseOptions
+        pathAsUri.path(),
+        pathAsUri.queryPairs.reduce((memo, param) => {
+          memo[param[0]] = param[1];
+          return memo;
+        }, {})
       );
```

Comment:
Sure. It could. I will need to add the babel plugin for object spread. But that is def more clear. 
I will try and update that.

Reply:
How can we use babel plugins? Using babel would make it a lot easier for something else that I'm working on.

As an aside, `Object.assign` can also be used in this case to avoid babel